### PR TITLE
feature/openshift-oauth-proxy

### PR DESCRIPTION
**Pull request by anothertobi**

### CODE REVIEW

File: openshift-oauth-proxy/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            - -provider=openshift
+            - -openshift-service-account={{ .Values.serviceaccount }}
+            - -cookie-secret-file=/etc/proxy/secrets/session_secret
+          {{- range $key, $value := .Values.extraArgs }}
+            - -{{ $key }}={{ $value }}
+          {{- end }}
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+          volumeMounts:
+          - mountPath: /etc/proxy/secrets
```
Comment (anothertobi) [inline openshift-oauth-proxy/templates/deployment.yaml]: Volume has to be configured and the respective secret created.

File: openshift-oauth-proxy/values.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+image:
+  repository: openshift/oauth-proxy
```
Comment (srueg) [inline openshift-oauth-proxy/values.yaml]: Please use the full name: `docker.io/openshift/oauth-proxy`

File: openshift-oauth-proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+The following table lists the configurable parameters of the OpenShift oauth-proxy chart. For defaults consult `values.yaml`
+
+| Parameter | Description
+| --- | ---
+| `image.repository` | Image repository
+| `image.tag` | Image tag
+| `image.pullPolicy` | Image pull policy
+| `extraArgs` | Extra arguments to pass to the oauth-proxy
+| `resources` | Define resource limits
+| `route.enabled` | Create route
```
Comment (srueg) [inline openshift-oauth-proxy/README.md]: This is not an available option

File: openshift-oauth-proxy/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+The following table lists the configurable parameters of the OpenShift oauth-proxy chart. For defaults consult `values.yaml`
+
+| Parameter | Description
+| --- | ---
+| `image.repository` | Image repository
+| `image.tag` | Image tag
+| `image.pullPolicy` | Image pull policy
+| `extraArgs` | Extra arguments to pass to the oauth-proxy
+| `resources` | Define resource limits
```
Comment (srueg) [inline openshift-oauth-proxy/README.md]: Please set default resources in `values.yaml`

File: openshift-oauth-proxy/values.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  tag: v1.1.0
+  pullPolicy: IfNotPresent
+serviceaccount:
+  name: openshift-oauth-proxy
+  annotations:
+    
+extraArgs:
+- upstream: http://localhost:9090
```
Comment (srueg) [inline openshift-oauth-proxy/values.yaml]: Please make the upstream and `openshift-sar` configurable in its own variable

File: openshift-oauth-proxy/values.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+---
+image:
+  repository: docker.io/openshift/oauth-proxy
+  tag: v1.1.0
+  pullPolicy: IfNotPresent
+serviceaccount:
+  name: openshift-oauth-proxy
+
+extraArgs:
+- openshift-delegate-urls: '{"/": {"resource": "namespaces", "verb": "get"}}'
```
Comment (srueg) [inline openshift-oauth-proxy/values.yaml]: Is this a sensible default? I don't really know what it does. Maybe it's better to comment it out, to show how an extra arg could be provided?

Comment (anothertobi) [reply openshift-oauth-proxy/values.yaml]: Commented it out and fixed some other things. Test was OK.

**Pull request merged**